### ISSUE TITLE
Add ONAP support to CNCF CI Cross Cloud Pipeline

### ISSUE THREAD

**Issue by lixuna** [crosscloudci/crosscloudci#4]
moved from https://github.com/crosscloudci/cross-cloud/issues/93

lixuna commented on Oct 11, 2017
As discussed in the ONAP + CNCF Cross Cloud Demo on Wed, Oct 11th, here is a ticket with next steps to adding ONAP Support to the pipeline:

Step 1: (start with one component)

Install one small component to see if they can get it to work (orchestration - made up of 2 containers, db and business logic)

Deploy 2 containers with helm

Deploy a simple robot test container (to validate that orchestration component is healthy)

Roger and the ONAP Team,

Let's use this ticket to outline the steps to take once the small component is ready for the CNCF WI Working Group.

Thank you for meeting with us today,
Lucina

---

 @obriensystems
  
obriensystems commented on Oct 11, 2017
ONAP tracking JIRA for integration work
https://jira.onap.org/browse/OOM-357

**Comment by obriensystems** [comment]
Lixuna,
   Hi, we are ready to assist you as required.
https://wiki.onap.org/display/~michaelobrien

**Comment by taylor** [comment]
Hi, @obriensystems.

I'm trying to do some initial verification of the v1.1.1 ONAP MSO container artifact.  I tried a docker pull on the one listed in a [gerrit commit](https://gerrit.onap.org/r/gitweb?p=oom.git;a=blob;f=kubernetes/mso/values.yaml;h=fd8b6690e8e9e185493b8b8c998bf53f2dbf81c9;hb=HEAD), `nexus3.onap.org:10001/openecomp/mso:v1.1.1`.  It's returning a 404 for the image though.

```
docker pull nexus3.onap.org:10001/openecomp/mso:v1.1.1
Pulling repository nexus3.onap.org:10001/openecomp/mso
Error: image openecomp/mso:v1.1.1 not found
```
Are these public?  If not can you assist with access?  Is there another issue?

Thanks,
Taylor

**Comment by obriensystems** [comment]
Hi, Michael here - sorry meetings straight all day - now free
Also, I will try to track this with my corporate account as part of ONAP under frank.obrien at amdocs.com
And we also have the ONAP discuss list - that is answered by most of the 300+ developers on the system
https://lists.onap.org/pipermail/onap-discuss/2018-February/
https://wiki.onap.org/display/DW/Mailing+Lists

What I will also do is link these issues into our JIRA system at jira.onap.org especially for the critical ones.
For deployment oriented JIRAs on the Kubernetes system - you may also enter ones at
https://jira.onap.org/secure/RapidBoard.jspa?rapidView=41
Hi, Michael here - sorry meetings straight all day - now free
Also, I will try to track this with my corporate account as part of ONAP under frank.obrien at amdocs.com
And we also have the ONAP discuss list - that is answered by most of the 300+ developers on the system
https://lists.onap.org/pipermail/onap-discuss/2018-February/
https://wiki.onap.org/display/DW/Mailing+Lists

What I will also do is link these issues into our JIRA system at jira.onap.org especially for the critical ones.
For deployment oriented JIRAs on the Kubernetes system - you may also enter ones at
https://jira.onap.org/secure/RapidBoard.jspa?rapidView=41

Also if you are presenting ONAP at a conference - we can act as a secondary tag team in the lead up to your talks - as we did for KubeCon last year and the F2F ONAP conference in Dec and as we will for the upcoming ONS conference in Mar.

OK, to the content

**Comment by obriensystems** [comment]
Yes, there does seem to be an issue - I noticed that nexus3 was restructured yesterday into v1/v2
on older systems with the last pull 2 months ago we are ok

My issue is why are we pulling MSO - as it was renamed SO - so I would expect a different name
but the manifest is correct
https://gerrit.onap.org/r/gitweb?p=integration.git;a=blob;f=version-manifest/src/main/resources/docker-manifest.csv;h=35e992adc0678ccd98328d33c9a5ffc88dbb4dfa;hb=refs/heads/master

ubuntu@ip-172-31-52-161:~/oom/kubernetes/oneclick$ docker pull nexus3.onap.org:10001/openecomp/mso:v1.1.1
Pulling repository nexus3.onap.org:10001/openecomp/mso
Error: image openecomp/mso:v1.1.1 not found


Doing this in parallel with some other work - so bear with me over the next hour as I perodically post to this message - will post when done

I expect the image dir has changed - and our existing poc CD because it does not pull very often on a clean VM yet - until I write the cloudformation - won't experience this - checking one of my other servers right - now


0:20:38 docker pull nexus3.onap.org:10001/openecomp/mso:v1.1.1
20:20:38 v1.1.1: Pulling from openecomp/mso
20:20:38 Digest: sha256:545b41e0d9dac152d44ba4cbced2dc2fb7b906f5fcd67fe0871c84342f13495a
20:20:38 Status: Image is up to date for nexus3.onap.org:10001/openecomp/mso:v1.1.1

On this CI server
ubuntu@ip-172-31-6-191:~$ sudo docker images | grep mso
nexus3.onap.org:10001/openecomp/mso                           v1.1.1               7758d7bfc845        2 months ago        1.57 GB

**Comment by obriensystems** [comment]
OK, forgot about the login first - tested a clean SPOT vm on AWS

Last login: Wed Feb  7 21:56:35 2018 from 32.60.102.40
ubuntu@ip-172-31-74-9:~$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
ubuntu@ip-172-31-74-9:~$ docker pull nexus3.onap.org:10001/openecomp/mso:v1.1.1
Pulling repository nexus3.onap.org:10001/openecomp/mso
Error: image openecomp/mso:v1.1.1 not found
ubuntu@ip-172-31-74-9:~$ docker login -u docker -p docker nexus3.onap.org:10001
Login Succeeded
ubuntu@ip-172-31-74-9:~$ docker pull nexus3.onap.org:10001/openecomp/mso:v1.1.1
v1.1.1: Pulling from openecomp/mso
13fe21ff56e6: Downloading [======================>                            ] 24.78 MB/55.54 MB
ce585474f014: Pull complete 
Digest: sha256:545b41e0d9dac152d44ba4cbced2dc2fb7b906f5fcd67fe0871c84342f13495a
Status: Downloaded newer image for nexus3.onap.org:10001/openecomp/mso:v1.1.1
ubuntu@ip-172-31-74-9:~$ docker images
REPOSITORY                            TAG                 IMAGE ID            CREATED             SIZE
nexus3.onap.org:10001/openecomp/mso   v1.1.1              7758d7bfc845        11 weeks ago        1.573 GB

Sorry a bit tired - thought you had a pull issue - but it is an anon request issue
You will need to login to docker:docker in order to pull these images


However try our release repo - you will have to retrofit all the values.yaml files though - there is an epic on parameterizing theses from the manifest - I will paste

https://hub.docker.com/r/onap/mso/tags/

ubuntu@ip-172-31-74-9:~$ docker pull onap/mso:v1.1.1
v1.1.1: Pulling from onap/mso
Digest: sha256:545b41e0d9dac152d44ba4cbced2dc2fb7b906f5fcd67fe0871c84342f13495a
Status: Downloaded newer image for onap/mso:v1.1.1
ubuntu@ip-172-31-74-9:~$ docker images
REPOSITORY                            TAG                 IMAGE ID            CREATED             SIZE
nexus3.onap.org:10001/openecomp/mso   v1.1.1              7758d7bfc845        11 weeks ago        1.573 GB
onap/mso                              v1.1.1              7758d7bfc845        11 weeks ago        1.573 GB

so you have 2 options - use the docker:docker login for pulling from nexus3 or use the release repo for anon access

consult the doc
https://wiki.onap.org/display/DW/ONAP+on+Kubernetes#ONAPonKubernetes-AutomatedInstallation

or the code

https://git.onap.org/oom/tree/kubernetes/oneclick/setenv.bash
ONAP_DOCKER_REGISTRY=${ONAP_DOCKER_REGISTRY:-nexus3.onap.org:10001}
ONAP_DOCKER_USER=${ONAP_DOCKER_USER:-docker}
ONAP_DOCKER_PASS=${ONAP_DOCKER_PASS:-docker}
/michael

https://wiki.onap.org/display/~michaelobrien

**Comment by obriensystems** [comment]
https://lists.onap.org/pipermail/onap-discuss/2018-February/008010.html

**Comment by obriensystems** [comment]
Your keynote at ONS just now 20180327 showed a heathcheck of 1/1 - it should be 23-27/30  It looks like you are still only running the mso pod of 2 containers - not the other 90.
Just curious - I'll ask at your booth.
I'll be demoing a full deployment of ONAP on Azure tomorrow for reference
https://onsna18.sched.com/event/Dm4T/onap-continuous-deployment-on-oom-kubernetes-via-public-cloud-michael-obrien-amdocs

/michael
